reco_utils/recommender/sar/sar_singlenode.py

```diff
@@ -319,7 +321,11 @@ def score(self, test, remove_seen=False):
             test_scores = test_scores.toarray()
 
         if self.normalize:
-            test_scores = rescale(test_scores, self.rating_scale_min, self.rating_scale_max, data_min=0)
+            counts = self.unity_user_affinity[user_ids, :].dot(self.item_similarity)
+            user_min_scores = np.tile(counts.min(axis=1)[:, np.newaxis], test_scores.shape[1]) * self.rating_min
+            user_max_scores = np.tile(counts.max(axis=1)[:, np.newaxis], test_scores.shape[1]) * self.rating_max
+            test_scores = (test_scores - user_min_scores) / (user_max_scores - user_min_scores)
+            test_scores = test_scores * (self.rating_max - self.rating_min) + self.rating_min
```

Comment:
This is actually identical to the `rescale` implementation I added. Can we keep that one and pass the per-user min/max scores of the "old" scale?

In that case `rescale` doesn't need to change at all, but would be called as such:

```py
counts = self.unity_user_affinity[user_ids, :].dot(self.item_similarity)
user_min_scores = np.tile(counts.min(axis=1)[:, np.newaxis], test_scores.shape[1]) * self.rating_min
user_max_scores = np.tile(counts.max(axis=1)[:, np.newaxis], test_scores.shape[1]) * self.rating_max
test_scores = rescale(test_scores, self.rating_min, self.rating_max, user_min_scores, user_max_scores)
```

Reply:
Ah I see. I wasn't sure if it worked with 2d datasets and 1d min/max arrays.
Yes please put that back in, and use it here.